internal/command/platform/status.go

```diff
@@ -2,31 +2,135 @@ package platform
 `
-		short = "Show current platform status"
+		short = "Show current platform status with an optional filter for maintenance or incidents in json mode (eg. incidents, maintenance)"
 	)
 
-	cmd = command.New("status", short, long, runStatus)
+	cmd = command.New("status [kind](optional)", short, long, runStatus)
+	cmd.Args = cobra.MaximumNArgs(1)
+	return
+}
+
+type Page struct {
+	ID        string    `json:"id,omitempty"`
+	Name      string    `json:"name,omitempty"`
+	Url       string    `json:"url,omitempty"`
+	UpdatedAt time.Time `json:"updated_at,omitempty"`
+}
 
-	cmd.Args = cobra.NoArgs
+type Status struct {
+	Description string `json:"description,omitempty"`
+	Indicator   string `json:"indicator,omitempty"`
+}
 
-	return
+type incidentUpdates struct {
+	Body      string    `json:"body,omitempty"`
+	Status    string    `json:"status,omitempty"`
+	CreatedAt time.Time `json:"created_at,omitempty"`
+	UpdatedAt time.Time `json:"updated_at,omitempty"`
+}
+
+type Incident struct {
+	ID              string          `json:"id,omitempty"`
+	Name            string          `json:"name,omitempty"`
+	Status          string          `json:"status,omitempty"`
+	CreatedAt       time.Time       `json:"port,omitempty"`
+	Impact          string          `json:"impact,omitempty"`
+	MonitoringAt    time.Time       `json:"monitoring_at,omitempty"`
+	PageID          string          `json:"page_id,omitempty"`
+	ResolvedAt      time.Time       `json:"resolved_at,omitempty"`
+	UpdatedAt       time.Time       `json:"updated_at,omitempty"`
+	IncidentUpdates incidentUpdates `json:"incident_updates,omitempty"`
+}
+
+type StatusPage struct {
+	Status Status
+	Page   Page
+}
+type Client struct {
+	baseUrl    *url.URL
+	httpClient *http.Client
+}
+
+func createClient(ctx context.Context, rawUrl string) (*Client, error) {
+	cleanedURL, _ := url.Parse(rawUrl)
+	logger := logger.MaybeFromContext(ctx)
+	httpClient, err := api.NewHTTPClient(logger, http.DefaultTransport)
+	if err != nil {
+		return nil, fmt.Errorf("can't setup HTTP client to %s: %w", rawUrl, err)
+	}
+
+	return &Client{
+		baseUrl:    cleanedURL,
+		httpClient: httpClient,
+	}, nil
 }
 
 func runStatus(ctx context.Context) error {
-	const url = "https://status.fly.io/"
+	const url = "https://status.fly.io"
+	var (
+		cfg               = config.FromContext(ctx)
+		getStatusEndpoint string
+		getStatusKind     = flag.FirstArg(ctx)
+	)
+
+	switch getStatusKind {
+	case "incidents":
+		getStatusEndpoint = "api/v2/incidents/unresolved.json"
+	case "maintenance":
+		getStatusEndpoint = "api/v2/scheduled-maintenances/active.json"
+	default:
+		getStatusEndpoint = "api/v2/status.json"
+	}
+
+	if cfg.JSONOutput {
+		client, _ := createClient(ctx, url)
+		req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("%s/%s", client.baseUrl, getStatusEndpoint), nil)
+		if err != nil {
+			log.Fatal(err)
+		}
+		var res *http.Response
+
+		if res, err = client.httpClient.Do(req); err != nil {
+			return err
+		}
```

Comment:
Looks like you can replace the whole thing with:

```suggestion
		httpClient, err := api.NewHTTPClient(logger.MaybeFromContext(ctx), http.DefaultTransport)
		if err != nil {
		    return err
		}
		res, err := httpClient.Get(url+getStatusEndpoint)
		if err != nil {
		    return err
		}
```

No need for `createClient` function and if concatenating url paths looks odd (it does), there is https://pkg.go.dev/net/url#JoinPath

Reply:
Oh neat!